aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp

```diff
@@ -204,14 +212,24 @@ namespace Aws
                 ByteBuffer digest(length);
                 memset(digest.GetUnderlyingData(), 0, length);
 
-                HMAC_CTX ctx;
-                HMAC_CTX_init(&ctx);
+                HMAC_CTX *ctx;
+#if OPENSSL_VERSION_NUMBER < 0x10100003L
+                HMAC_CTX _ctx;
+                ctx = &_ctx;
+                HMAC_CTX_init(ctx);
+#else
+                ctx=HMAC_CTX_new();
```

Comment:
Why are we now allocating things on the heap? Is this required for the latest OpenSSL?